PWG/EMCAL/READMEemcEmbedding.txt

```diff
@@ -382,10 +382,33 @@ embeddingHelper->SetAutoConfigureIdentifier("myEmbedding");
 embeddingHelper->SetNPtHardBins(11);
 ~~~
 
-Now just request the trains. Each one can be started with the desired settings, and they will automatically coordinate
-to determine which pt hard bins should be selected for each train. No need to manually change variables for each pt hard bin!
-
-With the above settings
+Now just request the trains. Each one can be started with the desired settings, and they will automatically
+coordinate to determine which pt hard bins should be selected for each train. No need to manually change variables
+for each pt hard bin! Note that there can be a race condition based on how quickly the test trains are started, so
+it is best to check the yaml pt hard map to ensure that one pt hard bin has successfully assigned to each train.
+
+# Optimization of Event Selection and Computing
+
+It is important to take care when applying event selection during embedding. For example, if the embedding helper
+is run with `AliVEvent::kAny`, but your task is run with `AliVEvent::kAnyINT`, some good embedded events will be
+missed because the physics selection of your task is more restrictive. There are two parts to solution to this issue.
+First, it is best to have the same collision candidates for the embedding helper and all other tasks.
+
+While this is a good start, it is not sufficient in all cases, such as selecting on centrality. To address this
+issue, the embedding helper allows for more complicated internal event selection via AliEventCuts (disabled by default).
+When enabled, The cuts can be configured through the YAML configuration for standard options such as centrality and
+z vertex, while more complicated manual configurations can be achieved by retrieving and configuring the members
+of the event cuts object. When an internal event is selected, `EmbeddedEventUsed()` will be true, allowing other tasks
+to only process events when this is the case. In doing so, the event selection that is applied in the embedding helper
+is effectively applied to all other tasks. Thus, be certain that any other event selection that you apply to other tasks
+is less restrictive than that in the embedding helper to ensure that no good embedded events are lost.
+
+To configure this mode, internal event selection must be enabled in the embedding helper via
+`SetUseManualInternalEventCuts(true)`, and then selection on the outcome from the embedding helper must be enabled
```

Comment:
I think you mean SetUseInternalEventSelection(true) here, yes?

Reply:
Yeah, you're right. Tab completion failure on my part. I'll update it